bool stop_matcher
bool reset_tracker
---
bool result
